File: src/parsing/operations/drop_binding.ts
import Bindings from "../../graph/bindings";
import Operation from "./operation";

/**
 * `DROP BINDING name`: remove a `LET`-bound entry (STATIC or eager)
 * from the {@link Bindings} singleton.  No-op if the binding does not
 * exist.
 */
class DropBinding extends Operation {
    private _name: string;

    constructor(name: string) {
        super();
        this._name = name;
    }

    public get name(): string {
        return this._name;
    }

    public async run(): Promise<void> {
        Bindings.getInstance().delete(this._name);
        await this.next?.run();
    }

    public get results(): Record<string, any>[] {
        return [];
    }
}

export default DropBinding;

File: flowquery-py/src/parsing/operations/drop_binding.py
"""DROP BINDING operation."""

from typing import Any, List

from ...graph.bindings import Bindings
from .operation import Operation


class DropBinding(Operation):
    """``DROP BINDING name``: remove a LET-bound entry (STATIC or
    eager) from the :class:`Bindings` singleton.  No-op if the binding
    does not exist.
    """

    def __init__(self, name: str):
        super().__init__()
        self._name = name

    @property
    def name(self) -> str:
        return self._name

    async def run(self) -> None:
        Bindings.get_instance().delete(self._name)
        if self.next:
            await self.next.run()

    @property
    def results(self) -> List[Any]:
        return []
